src/components/bmfont-text.js

```diff
@@ -0,0 +1,261 @@
+var registerComponent = require('../core/component').registerComponent;
+var THREE = require('../lib/three');
+
+var createTextGeometry = require('three-bmfont-text');
+var loadBMFont = require('load-bmfont');
+var path = require('path');
+var assign = require('object-assign');
+var createSDF = require('three-bmfont-text/shaders/sdf');
+var createMSDF = require('three-bmfont-text/shaders/msdf');
+var createBasic = require('three-bmfont-text/shaders/basic');
+
+var alignments = ['left', 'right', 'center'];
+var anchors = ['left', 'right', 'center', 'align'];
+var baselines = ['top', 'center', 'bottom'];
+
+var DEFAULT_WIDTH = 1; // 1 matches other AFRAME default widths... 5 matches prior bmfont examples etc.
+
+var FONT_BASE_URL = 'https://cdn.rawgit.com/fernandojsg/aframe-bmfont-component/5c88edf40bbc88b336d7db6f040c3ae0d2f65aff/fonts/';
+var fontMap = {
+  'default': FONT_BASE_URL + 'DejaVu-sdf.fnt',
+  'Aileron-Semibold': FONT_BASE_URL + 'Aileron-Semibold.fnt',
+  'DejaVu': FONT_BASE_URL + 'DejaVu-sdf.fnt',
+  'Exo2Bold': FONT_BASE_URL + 'Exo2Bold.fnt',
+  'Exo2SemiBold': FONT_BASE_URL + 'Exo2SemiBold.fnt',
+  'KelsonSans': FONT_BASE_URL + 'KelsonSans.fnt',
+  'Monoid': FONT_BASE_URL + 'Monoid.fnt',
+  'SourceCodePro': FONT_BASE_URL + 'SourceCodePro.fnt',
+  'mozillavr': FONT_BASE_URL + 'mozillavr.fnt'
+};
+
+module.exports.Component = registerComponent('bmfont-text', {
+  schema: {
+    // scale is now determined by width and wrappixels/wrapcount... scale: {default: 0.003},
+    font: {default: ''},
+    tabSize: {default: 4},
+    anchor: {default: 'center', oneOf: anchors}, // center default to match primitives like plane; if 'align', null or undefined, same as align
+    baseline: {default: 'center', oneOf: baselines},
+    text: {type: 'string'},
+    width: {type: 'number'}, // use AFRAME units i.e. meters, not arbitrary numbers... // default to geometry width, or if not present then DEFAULT_WIDTH
+    height: {type: 'number'}, // use AFRAME units i.e. meters, not arbitrary numbers... // no default, will be populated at layout
+    align: {type: 'string', default: 'left', oneOf: alignments},
+    letterSpacing: {type: 'number', default: 0},
+    lineHeight: {type: 'number'},  // default to font's lineHeight value
+    fnt: {type: 'string', default: 'default'},
+    fntImage: {type: 'string'}, // default to fnt but with .fnt replaced by .png
+    mode: {default: 'normal', oneOf: ['normal', 'pre', 'nowrap']},
+    color: {type: 'color', default: '#000'},
+    opacity: {type: 'number', default: '1.0'},
+    type: {default: 'SDF', oneOf: ['SDF', 'basic', 'MSDF']},
```

Comment:
if we are not worried about compatibility with the prior component, sure
but here I think the intent was to note the type of font (basic, signed distance field, multichannel signed distance field) so actually `type` is more appropriate than directly referencing `shader` IMO